### PR TITLE
cross-build for sbt 0.13 and sbt 1.0

### PR DESCRIPTION
**Pull request by jeffreyolchovy**
Tested with:
```
[sbt-header]> ;^clean ;^compile ;^test ;^scripted
```

All succeeded.

### CODE REVIEW

File: project/build.properties [hunk 1]
```diff
@@ -1 +1 @@
-sbt.version = 1.0.0
+sbt.version = 0.13.16
```
Comment (hseeberger) [inline project/build.properties]: No, I want the main sbt version to be 1.0.

Comment (jeffreyolchovy) [inline project/build.properties]: No worries, feel free to close, just opened this as an example for a dialogue in the sbt gitter channel.

Comment (nrinaudo) [inline project/build.properties]: Just for the record though, I understand this is the only way to cross-build. You need to be running 0.13.16.

Comment (hseeberger) [inline project/build.properties]: Nope, works for 1.0.0, too.

Comment (jeffreyolchovy) [inline project/build.properties]: yeah, just pushed another small change so sbt 1.0.0 is the main version

File: build.sbt [hunk 2]
```diff
@@ -45,16 +45,22 @@ lazy val commonSettings =
       "-unchecked",
       "-deprecation",
       "-language:_",
-      "-target:jvm-1.8",
       "-encoding", "UTF-8"
     ),
+    scalacOptions += {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((2, 12)) => "-target:jvm-1.8"
+        case _ => "-target:jvm-1.7"
```
Comment (hseeberger) [inline build.sbt]: Please align arrows.

File: build.sbt [hunk 3]
```diff
@@ -45,16 +45,22 @@ lazy val commonSettings =
...
+      }
+    },
     unmanagedSourceDirectories.in(Compile) := Seq(scalaSource.in(Compile).value),
     unmanagedSourceDirectories.in(Test) := Seq(scalaSource.in(Test).value),
     shellPrompt in ThisBuild := { state =>
       val project = Project.extract(state).currentRef.project
       s"[$project]> "
     },
     sbtPlugin := true,
+    crossSbtVersions := Seq("0.13.16", "1.0.0"),
```
Comment (hseeberger) [inline build.sbt]: 1.0.0 first, please.

**Comment by nrinaudo** [comment]
Wouldn't you want to cross-build in `.travis.yml` though? Something like this? https://travis-ci.org/nrinaudo/sbt-header

(I'm just talking about the `.travis.yml`, this PR is of an obviously higher caliber than what I'd bee doing)

**Comment by hseeberger** [comment]
@nrinaudo yes, of course, good catch! And no, your PR is even richer ;-)

**Comment by hseeberger** [comment]
@nrinaudo do we really need this Travis voodoo magic? Ain't sbt smart enough to figure out the Scala version it needs?

**Comment by jeffreyolchovy** [comment]
and btw, there's nothing really here content-wise, so I cede the change to you guys .. like I said, feel free to close in favor of @nrinaudo's work, was just trying to help out as I was doing similar work these past few weeks. best of luck!

**Comment by hseeberger** [comment]
@jeffreyolchovy don't worry, well get there.

**Comment by nrinaudo** [comment]
Honestly, I ripped that directly from Eugene Yokota's `sbt-unidoc`, figuring he knew what he was doing. This whole matrix thing is not something I pretend to understand :)

We can't really let SBT decide on the version of java it needs though, unfortunately. SBT 0.13.16 is quite happy to run on 1.6, but you're using scalafmt, which needs 1.8+...

**Comment by nrinaudo** [comment]
@jeffreyolchovy sorry, I didn't mean to try and steal your thunder. I don't even *have* a PR ready, I'm just playing with things to see whether I can get them to work, if not locally, at least on Travis. Your help is very much appreciated, and I certainly learned loads here!

**Pull request merged**